### ISSUE TITLE
Distributor: Events are lost if service container is not ready yet

### ISSUE THREAD

**Issue by bacherfl** [keptn/keptn#7290]
When testing the zero downtime capabilities of Keptn, i came across several occasions where e.g. an `evaluation.triggered` event sent by the shipyard controller is not processed by the lighthouse service, when there is an upgrade off Keptn (or the lighthouse service) going on. 
The upgrade strategy for the lighthouse service was as follows:

```
strategy:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
     rollingUpdate:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
       maxSurge: 1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
       maxUnavailable: 0                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
     type: RollingUpdate
```

So this should ensure that there is always at least one instance of the lighthouse service pod that is able to receive traffic.
I also noticed that the `evaluation.triggered` event did get through to one of the distributors, but apparently the lighthouse-service container was not ready at this point, which lead to the event being lost.

This scenario can be reproduced relatively consistently (it does not always happen, but still often enough to cause problems):

1. Continuously sequences for a service (e.g. evaluation sequences)

2.Update the image of the lighthouse-service, e.g. using the following command:

```
kubectl -n keptn set image deployment.v1.apps/lighthouse-service lighthouse-service=keptndev/lighthouse-service:0.14.0-dev-PR-7266.202203280650
```

Then you should see some sequences in the bridge which have been triggered, and if the problem happened, the currently active sequence will be stalled because of no `evaluation.started` event being received. Eventually, this sequence will time out and be aborted by the shipyard controller 


Another case where the incoming evaluation.triggered events were not handled is during the termination of the lighthouse service container. During this period, the distributor will still get events, but those are not handled by the lighthouse service anymore, as the logs below suggest:

```
lighthouse-service time="2022-03-28T14:19:33Z" level=fatal msg="Container termination triggered, waiting for graceful shutdown"                                                                                                                            │
│ distributor time="2022-03-28T14:19:35Z" level=info msg="Received a message for topic [sh.keptn.event.evaluation.triggered]\n"                                                                                                                              │
│ distributor time="2022-03-28T14:19:35Z" level=info msg="Sending CloudEvent with ID 067dcf91-56d9-4216-a96e-80bd2b0f0a0a to 127.0.0.1"                                                                                                                      │
│ distributor time="2022-03-28T14:19:37Z" level=info msg="Received a message for topic [sh.keptn.event.evaluation.triggered]\n"                                                                                                                              │
│ distributor time="2022-03-28T14:19:37Z" level=info msg="Sending CloudEvent with ID 37cadc36-1065-4b66-b7b2-c6571ae2aad6 to 127.0.0.1"                                                                                                                      │
│ distributor time="2022-03-28T14:19:39Z" level=info msg="Received a message for topic [sh.keptn.event.evaluation.triggered]\n"                                                                                                                              │
│ distributor time="2022-03-28T14:19:39Z" level=info msg="Sending CloudEvent with ID cf37fed9-c245-40fb-949c-bb5eb7f7b2fe to 127.0.0.1"                                                                                                                      │
│ distributor time="2022-03-28T14:19:45Z" level=info msg="Received a message for topic [sh.keptn.event.evaluation.triggered]\n"                                                                                                                              │
│ distributor time="2022-03-28T14:19:45Z" level=info msg="Sending CloudEvent with ID 8373579b-5ea0-48da-aa33-8a9db39c4248 to 127.0.0.1"                                                                                                                      │
│ distributor time="2022-03-28T14:19:53Z" level=info msg="Terminating event forwarder"                                                                                                                                                                       │
│ distributor time="2022-03-28T14:19:53Z" level=info msg="Terminating UniformLogger"                                                                                                                                                                         │
│ distributor time="2022-03-28T14:19:53Z" level=info msg="Unsubscribed from NATS topic: sh.keptn.event.evaluation.triggered"                                                                                                                                 │
│ distributor time="2022-03-28T14:19:53Z" level=info msg="Unsubscribed from NATS topic: sh.keptn.event.get-sli.finished"                                                                                                                                     │
│ distributor time="2022-03-28T14:19:53Z" level=info msg="Unsubscribed from NATS topic: sh.keptn.event.monitoring.configure"                                                                                                                                 │
│ distributor time="2022-03-28T14:19:53Z" level=info msg="Terminating NATS event receiver
```

**Comment by thisthat** [comment]
This will be fixed for Keptn core services with #7435

**Comment by thisthat** [comment]
https://github.com/keptn/keptn/issues/7435 has been closed. We can, in turn, close this issue :)
